pkg/security/tests/filters_test.go

```diff
@@ -148,6 +149,9 @@ func TestOpenLeafDiscarderFilter(t *testing.T) {
 		t.Fatalf("event inode: %d, parent inode: %d, error: %v", inode, parentInode, err)
 	}
 
+	// The func pushing the discarder needs to complete before testing the discarder
+	time.Sleep(time.Second * 2)
```

Comment:
I'm glad that we have the same behavior for this test and the TestDiscarderRetentionFilter one.
We should try to address them in the same way, I was first wondering if to fix them with a sleep addition like you did, but after some discussion we have decided to rework the code part that is the root cause of the race.